### ISSUE TITLE
Bump cgmath to 0.18.0

### ISSUE THREAD

**Issue by danakj** [DataAnalysisCosby/mgf#12]
The new cgmath has some nice helpers added.

The CHANGELOG.md specifies a breaking change however: https://github.com/rustgd/cgmath/blob/master/CHANGELOG.md

> Breaking: Quaternion memory layout changed to [x, y, z, w]. The From and Into impls for [S; 4] and (S, S, S, S) have been changed accordingly.

The data has moved from

  pub struct Quaternion {
      pub s: S,
      pub v: Vector3,
  }

To

  pub struct Quaternion {
      pub v: Vector3,
      pub s: S,
  }

I'm not sure if mgf depends on from/to [S;4] or (S,S,S,S). It seems unlikely to me, since those are mostly for interfacing to something like OpenGL, but I wanted to raise the issue here.

**Comment by maplant** [comment]
Hmmm it should be pretty easy to bump this, however there are a few externalities in doing so that complicate things:

- bumping cgmath, since this is a re-export, counts as an API breaking change
- I believe that GJK is flat out broken here, as I have a lot of the functionality extracted from MGF and used for other projects, and that functionality has been substantially modified. I bring this up because I don't think I can in good conscious update this repo without fixing these problems, and that would take quite a bit of work.

In short, when I have the bandwidth I will try to do this, but it might take a bit. In the mean time, I advise against using MGF for anything serious, and if you want to use the continuous collision detection algorithms go ahead and take them or use them as reference.
